pkg/monitoring/metrics_labeler.go

```diff
@@ -0,0 +1,33 @@
+package monitoring
+
+type MetricsLabeler struct {
+	Labels map[string]string
+}
+
+func NewMetricsLabeler() MetricsLabeler {
+	return MetricsLabeler{Labels: make(map[string]string)}
+}
+
+// With adds multiple key-value pairs to the CustomMessageLabeler for transmission With SendLogAsCustomMessage
```

Comment:
/nit This is my bad from when I implemented this in core
```suggestion
// With adds multiple key-value pairs to the MetricsLabeler to eventually be consumed by a Beholder metrics resource
```